rfcs/text/0008_service_status.md

```diff
@@ -0,0 +1,251 @@
+- Start Date: 2020-03-07
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+A set API for describing the current status of a system (Core service or plugin)
+in Kibana.
+
+# Basic example
+
+```ts
+core.status.set({
+  level: ServiceStatusLevel.degraded,
+  summary: `APM indices cannot be created`,
+});
+```
+
+# Motivation
+
+Kibana should do as much possible to help users keep their installation in a working state. This includes providing as much detail about components that are not working as well as ensuring that failures in one part of the application do not block using other portions of the application.
+
+In order to provide the user with as much detail as possible about any systems that are not working correctly, the status mechanism should provide excellent defaults in terms of expressing relationships between services and presenting detailed information to the user.
+
+# Detailed design
+
+## Types
+
+```ts
+/**
+ * The current status of a service at a point in time.
+ */
+interface ServiceStatus {
+  /** The current availability level of the service. */
+  level: ServiceStatusLevel;
+  /** A high-level summary of the service status. */
+  summary?: string;
```

Comment:
Is an optional `summary` enough or should we have something like a `cause` field that is mandatory when the level !== `available` ?

Reply:
Makes sense to me 👍